### PR TITLE
Add GitHub provider

### PR DESCRIPTION
**Pull request by ataylorme**

### CODE REVIEW

**Comment by ataylorme** [comment]
Tested well on https://github.com/ataylorme/build-tools-gatsby-wp-poc-2019-09-18

File: src/ServiceProviders/RepositoryProviders/GitHub/GitHubProvider.php [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      $postData['base'] = $project_details['default_branch'];
+    }
+
+    if(!empty($options['draft'])) {
+      $postData['draft'] = boolval($options['draft']);
+    }
+
+    // GitHub doesn't have reviewers on pull request creation
+    // See https://developer.github.com/v3/pulls/#create-a-pull-request
+    unset($options['reviewers']);
```
Comment (aaronbauman) [inline src/ServiceProviders/RepositoryProviders/GitHub/GitHubProvider.php:61]: Should we also update https://github.com/pantheon-systems/terminus-clu-plugin/blob/master/src/Commands/PullRequestCreateCommand.php to spell out which options are available for which providers?

Comment (ataylorme) [reply src/ServiceProviders/RepositoryProviders/GitHub/GitHubProvider.php:61]: Updated in [`0e04621`](https://github.com/pantheon-systems/terminus-clu-plugin/pull/2/commits/0e046216b1377ff8b55cad279588f10c2d4acf38)

**Review by aaronbauman — APPROVED** [review]
LGTM

**Pull request merged**